sdk/eventhub/Azure.Messaging.EventHubs.Shared/src/Processor/PartitionLoadBalancer.cs

```diff
@@ -0,0 +1,439 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using Azure.Core;
+using Azure.Messaging.EventHubs.Core;
+using Azure.Messaging.EventHubs.Errors;
+
+namespace Azure.Messaging.EventHubs.Processor
+{
+    internal class PartitionLoadBalancer
+    {
+        /// <summary>The random number generator to use for a specific thread.</summary>
+        private static readonly ThreadLocal<Random> RandomNumberGenerator = new ThreadLocal<Random>(() => new Random(Interlocked.Increment(ref s_randomSeed)), false);
+
+        /// <summary>The seed to use for initializing random number generated for a given thread-specific instance.</summary>
+        private static int s_randomSeed = Environment.TickCount;
+
+        /// <summary>
+        ///   Interacts with the storage system with responsibility for creation of checkpoints and for ownership claim.
+        /// </summary>
+        ///
+        private readonly PartitionManager StorageManager;
```

Comment:
We'd have to move `PartitionManager` into the shared area as well, which we 're likely to need regardless.   I'm guessing `Shared` isn't able to compile in the current state.

Reply:
As I recall, Shared is a shared project (shproj) - does it compile on its own? 
The Processor project builds and runs tests successfully though.

Reply:
Oh, I see now if I build the .sln

Reply:
Yeah, the goal is for the shared code to be fully stand-alone, including responsibility for managing its own tests via the associated test project.   When one of the other projects pulls in a bundle, it should have confidence that the code is stable.  

Because the test project pulls in all of the bundles from the shared code, anything in there needs to be compilable.   Since we'd want to have load balancing testable, we'd need to pull the `PartitionManager` in for that anyway.